clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/InAppController.java

```diff
@@ -706,7 +706,7 @@ private static void showInApp(Context context, final CTInAppNotification inAppNo
             return;
         }
 
-        boolean isHtmlType = inAppNotification.getType().equals(Constants.KEY_CUSTOM_HTML);
+        boolean isHtmlType = inAppNotification.getType() != null && inAppNotification.getType().equals(Constants.KEY_CUSTOM_HTML);
```

Comment:
inAppNotification.getType() -> extract once

Reply:
Resolved. Just curious if it only improves readability or does it improve performance as well?

Reply:
just from a thread safety pov, had seen some problem in other parts of code.